src/pudl/transform/phmsagas.py

```diff
@@ -219,10 +318,31 @@ def _check(df):
         ):
             error = "Percentage of rows with negative percent_unaccounted_for_gas values: {negative_percentage:.2f}"
             logger.info(error)
+            return AssetCheckResult(passed=False, metadata={"errors": error})
+
+        return AssetCheckResult(passed=True)
 
-        if error:
+    @asset_check(asset=spec.asset, blocking=True)
```

Comment:
Not entirely following the logic here - the asset check will run on the `df` after you've dropped these rows in the transform step to create a unique PK, so the `non_unique_groups` returned in this asset check should always be 0. Rather than writing an asset check here, we'll want to actually embed this assertion in the code itself. 

Could be as simple as something like this in the actual filtering functions you've written:
```
assert unfiltered_df.shape[0] - filtered_df.shape[0]  <=3, "Dropped more than expected number of rows due to non-unique groups. Rows dropped: {dropped_rows}
```